docs/rst/manual/reference/configuration/magic_strings.rst

```diff
@@ -0,0 +1,259 @@
+
+Magic strings
+=============
+
+ERT exposes some magic strings that can be used in your configuration file.
+These strings will at runtime be replaced with corresponding values, and
+can be useful if you need to use information such as file paths or experiment
+information such as realization or iteration number in your workflows or jobs.
+
+Example:
+::
+
+   YOUR_WORKFLOW_JOB <RUNPATH_FILE> <CONFIG_PATH>
+
+Will be replaced with something similar to the following at runtime
+
+::
+
+   YOUR_WORKFLOW_JOB /path/to/the/runpath/file/.ert_runpath_list /path/to/config/
+
+In the following we separate between :ref:`experiment constants <ms_exp_constants>`
+and :ref:`experiment variables <ms_exp_variables>`. Experiment are magic strings that
+do not change based on the iteration or realization number, while experiment variables
+will typically be replaced with different values depending on the current iteration or
+realization index.
+
+**It should also be noted that experiment variables can only be used with forward model jobs
+and not with workflows.**
+
+
+List of keywords
+----------------
+
+
+* Experiment Constants
+
+   * :ref:`RUNPATH_FILE <ms_runpath_file>`
+   * :ref:`CONFIG_PATH <ms_config_path>`
+   * :ref:`CONFIG_FILE <ms_config_file>`
+   * :ref:`CONFIG_FILE_BASE <ms_config_file_base>`
+   * :ref:`CWD <ms_cwd>`
+
+* Experiment Variables
+
+   * :ref:`IENS <ms_iens>`
```

Comment:
In some ways I would like to separate between `IENS`, `ITER`, `GEO_ID` and the rest. As the first are actively changed by ERT, always! While the later can change over an experiment in a very specific manner if the user parameterises it by one of the first three.

Could we split them and name them `Active` and `Passive`? Or perhaps borrow from statistics and name the first `Experiment Variables` and the later `Experiment Dependent Variables`?

Reply:
They are separated already (`Experiment Constants` vs Experiment Variables), and the current split is made based on where in the code they are modified (`Experiment variables` are modified with the `run_args`) (see below). I agree that `Experiment Variables` and the later `Experiment Dependent Variables` might be better names! The only odd keyword in this list is `ECLBASE`, which I though was constant for an experiment?  

Experiment Constants

- RUNPATH_FILE
- CONFIG_PATH
- CONFIG_FILE
- CONFIG_FILE_BASE
- CWD

Experiment Variables

- IENS
- ITER
- ECLBASE/ECL_BASE
- GEOID
- RUNPATH
- NUM_CPU

Reply:
I know. So does `RUNPATH` ;) And while `ITER` is actively changed by ERT itself, ECLBASE is allowed to change due to it being ran through substitution. Hence, my suggestion was whether we should make it into `Experiment Constants`, `Experiment dependent variables` and `Experiment variables` where it is only the `Experiment variables` that are actively changed during an experiment by ERT. The `Experiment dependent variables` might be impacted by this change and the `Experiment Constants` are promised to remain the same for the entire run 🤷

Reply:
Ahh, I misunderstood you comment! I agree to those changes!